src/main/scala/nl/knaw/dans/easy/bagindex/package.scala

```diff
@@ -96,8 +66,7 @@ package object bagindex {
     def asScala: Option[T] = if (opt.isPresent) Some(opt.get()) else None
   }
 
-
-  // TODO: WILL BE IN dans-scala-lib
+  // TODO: will NOT be in dans-scala-lib, because it actually doesn't work correctly...
```

Comment:
You're right. Kind of looks like Fermat's last theorem now. To be clear, it works here but it doesn't in the general case. I tested that when I wrote easy-metadata-to-datacite. It has to do with the evaluation order. Since it is a stream, you want FULLY lazy evaluation. Turns out, you can't get that with this. To get a Try, you need to evaluate the stream, which you don't want. Instead, the stream inside the Success must be fully evaluated. If you really want laziness and error control, you need an Observable. However, I don't know yet how to work with Observable and a REST API. Once we figured that out, we can remove this extension method.
To be clear, again, it works here, but it doesn't in general. That's why we'll not add it to the library.